codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -228,14 +251,20 @@ public void dataWriteShouldSucceed() throws Exception {
 
     @Test
     public void dataWriteShouldHalfCloseStream() throws Exception {
-        reset(future);
+        Http2Connection connection = new DefaultHttp2Connection(true);
```

Comment:
Hmm ... would it make things easier if we used a real connection in the default encoder? Then you wouldn't need to re-create the encoder here.